java/src/main/java/org/rocksdb/RocksDB.java

```diff
@@ -61,35 +63,39 @@ public static synchronized void loadLibrary() {
    *     of a library.
    */
   public static synchronized void loadLibrary(final List<String> paths) {
-    for (CompressionType compressionType : CompressionType.values()) {
-      if (compressionType.equals(CompressionType.NO_COMPRESSION)) {
-        continue;
+    if(!libraryLoaded) {
```

Comment:
`synchronized` in this case would be against the class because the methods are static. I somehow missed those `synchronized` keywords, I will send an update in a moment that hopefully does better.